internal/weldr/api.go

```diff
@@ -1769,16 +1776,33 @@ func (api *API) depsolveBlueprint(bp *blueprint.Blueprint, clean bool) ([]rpmmd.
 			specs[i] += "-*-*.*"
 		}
 	}
+	excludeSpecs := []string{}
+	if outputType != "" {
+		// When the output type is known, include the base packages in the depsolve
+		// transaction.
+		packages, excludePackages, err := api.distro.BasePackages(outputType, arch)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		specs = append(specs, packages...)
+		excludeSpecs = append(excludePackages, excludeSpecs...)
+	}
 
-	var repos []rpmmd.RepoConfig
-	for _, repo := range api.distro.Repositories(api.arch) {
-		repos = append(repos, repo)
+	packages, checksums, err := api.rpmmd.Depsolve(specs, excludeSpecs, repos, api.distro.ModulePlatformID(), clean)
+	if err != nil {
+		return nil, nil, nil, err
 	}
-	for _, source := range api.store.GetAllSources() {
-		repos = append(repos, source.RepoConfig())
+
+	buildPackages := []rpmmd.PackageSpec{}
+	if outputType != "" {
+		buildSpecs, err := api.distro.BuildPackages(arch)
+		if err != nil {
+			return nil, nil, nil, err
```

Comment:
Do we have a reasonable way to tell the user what happened here? Was it possible to have such errors in Weldr?

Reply:
This would be an programming error on our side. We should `panic()` or at most return `InternalServerError`.